src/Orleans.Core.Abstractions/Metadata/FeaturePopulatorAttribute.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace Orleans.Metadata
+{
+    /// <summary>
+    /// Defines a feature populator for this assembly.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public sealed class FeaturePopulatorAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FeaturePopulatorAttribute"/> class.
+        /// </summary>
+        /// <param name="populatorType">The feature populator type.</param>
+        public FeaturePopulatorAttribute(Type populatorType)
+        {
+            this.PopulatorType = populatorType;
```

Comment:
A check for ```IFeaturePopulator``` would make sense here IMO, just to be foolproof.

Reply:
essentially: `populatorType.GetInterfaces().Any(iface => iface.IsConstructedGenericType && typeof(IFeaturePopulator<>).IsAssignableFrom(iface.GetGenericTypeDefinition()))`;?